src/Microsoft.Performance.SDK/Extensibility/ICookedDataRetrievalExtensions.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Performance.SDK.Extensibility.DataCooking;
+
+namespace Microsoft.Performance.SDK.Extensibility
+{
+    /// <summary>
+    ///     Additional functionality for the <see cref="ICookedDataRetrieval"/> interface.
+    /// </summary>
+    public static class ICookedDataRetrievalExtensions
+    {
+        /// <summary>
+        ///     Retrieves data by name, typecast to the given type.
+        ///     The caller is coupled to the data extension and expected to know the
+        ///     data type.
+        /// </summary>
+        /// <typeparam name="T">
+        ///     Type of the data to retrieve.
+        /// </typeparam>
+        /// <param name="retrieval">
+        ///     Instance on which to act.
+        /// </param>
+        /// <param name="dataOutputPath">
+        ///     Unique identifier for the data to retrieve.
+        /// </param>
+        /// <param name="result">
+        ///     The uniquely identified data.
+        /// </param>
+        /// <exception cref="InvalidOperationException">
+        ///     The specified <paramref name="dataOutputPath"/> was not found.
+        /// </exception>
+        /// <exception cref="InvalidCastException">
+        ///     The value at <paramref name="dataOutputPath"/> cannot be cast to <typeparamref name="T"/>.
+        /// </exception>
+        public static void QueryOutput<T>(
```

Comment:
Why does this extension exist? It returns void and uses an out parameter. Do we need it?

Reply:
It makes it so that you don't need to specify the type T when making the call. For example, this works...

QueryOutput(path, out bool result);